File: app/src/main/java/de/slg/datadome/Filter.java
package de.slg.datadome;

import java.util.List;
import java.util.ListIterator;

public class Filter {

    public List<Article> filtern (List<Article> articleList, List<Short> category)
    {
        List<Article> filtertList = null;
        int numbersOfCategories = category.size();
        int i = articleList.size();
        while (i>0)
        {
            Article current = articleList.get(i);
            List<Short> currentCategories = current.getCategoryIds();
            int b = currentCategories.size();

            while (numbersOfCategories>0)
            {
                while(b>0)
                {
                    if (category.get(numbersOfCategories).equals(currentCategories.get(b)))
                    {
                      if (!current.getAbstractText().equals(""))
                      {filtertList.add(current);}
                    }
                }
            }

            i--;
        }

        return filtertList;
    }
}

File: app/src/main/java/de/slg/datadome/Filter.kt
package de.slg.datadome

import java.util.*

fun filterCategory(locations: List<MapLocation>, categories: List<Short>): List<MapLocation> {

    val filteredList = mutableListOf<MapLocation>()

    for (loc in locations) {
        for (s in categories) {
            if (loc.categoryId == s) {
                filteredList.add(loc)
                break
            }
        }
    }

    return filteredList
}

fun filterTime(articleList: List<MapLocation>, from: Date, to: Date): List<MapLocation> {
    val filteredList: MutableList<MapLocation> = mutableListOf()
    for (current in articleList) {
        if (current.dates.isEmpty())
            filteredList.add(current)
        else {
            for (d in current.dates) {
                if (d.from.before(from) && d.to!!.after(to))
                    filteredList.add(current)
                else if (d.from.before(from) && d.to == null)
                    filteredList.add(current)

            }
        }

    }

    return filteredList
}